ladybug/src/main/java/nl/nn/ibistesttool/larva/ConvertToLarvaAction.java

```diff
@@ -0,0 +1,322 @@
+						skipUntilEndOfSender = false;
+					}
+				} else if (checkpoint.getType() == Checkpoint.TYPE_STARTPOINT && checkpoint.getName().startsWith("Sender ")) {
+					if (!allowedSenders.contains(checkpoint.getSourceClassName())) {
+						//If sender should be stubbed:
+						String senderName = checkpoint.getName().substring(7);
+						String senderProperty = "stub." + senderName;
+						scenarioProperties.setProperty("step" + ++current_step_nr + "." + senderProperty + ".read", scenarioDirPrefix + stepPadding(current_step_nr) + "-stub-" + senderName + "-in.xml");
+						createInputOutputFile(scenarioDir, current_step_nr, "stub", senderName, true, checkpoint.getMessage());
+						scenarioProperties.setProperty("step" + ++current_step_nr + "." + senderProperty + ".write", scenarioDirPrefix + stepPadding(current_step_nr) + "-stub-" + senderName + "-out.xml");
+
+						String serviceName = "testtool-Call" + senderName;
+						String existingStubName = existingStubs.get(serviceName.toLowerCase());
+						if (!senderProperty.equals(existingStubName)) {
+							existingStubs.put(serviceName.toLowerCase(), senderProperty);
+							commonProperties.setProperty(senderProperty + ".className", "nl.nn.adapterframework.receivers.JavaListener");
+							commonProperties.setProperty(senderProperty + ".serviceName", serviceName);
+
+							if (existingStubName != null) {
+								commonProperties.removeProperty(existingStubName + ".className");
+								commonProperties.removeProperty(existingStubName + ".serviceName");
+								try {
+									replaceStubName(scenario.getParent(), existingStubName, senderProperty);
+								} catch (IOException e) {
+									errors.add("Could not replace old stub name [" + existingStubName + "] with new stub name [" + senderProperty + "]");
+								}
+							}
+						}
+
+						skipUntilEndOfSender = true;
+						skipUntilEndOfSenderName = senderName;
+						skipUntilEndOfSenderLevel = checkpoint.getLevel() + 1;
+					}
+				} else if (checkpoint.getLevel() == 1 && checkpoint.getType() == Checkpoint.TYPE_INPUTPOINT) {
+					//SessionKey for listener found
+					String sessionKeyName = checkpoint.getName().substring(11);
+					if (!ignoredSessionKeys.contains(sessionKeyName)) {
+						scenarioProperties.setProperty(adapterProperty + ".param" + paramI + ".name", sessionKeyName);
+						scenarioProperties.setProperty(adapterProperty + ".param" + paramI + ".value", checkpoint.getMessage());
+						paramI++;
+					}
+				}
+			}
+			scenarioProperties.setProperty("step" + ++current_step_nr + "." + adapterProperty + ".read", scenarioDirPrefix + stepPadding(current_step_nr) + "-adapter-" + adapterName + "-out.xml");
+			createInputOutputFile(scenarioDir, current_step_nr, "adapter", adapterName, false, checkpoints.get(checkpoints.size() - 1).getMessage());
+
+			try {
+				scenarioWriter = new OutputStreamWriter(Files.newOutputStream(scenarioFile.toPath()), StandardCharsets.UTF_8);
+				scenarioProperties.store(scenarioWriter, null);
```

Comment:
Nu worden de writers in geheugen gehouden gedurende de hele methode, je kan ze beter naar het try block verplaatsen, en nog beter, try with resources, dan doet hij ook een close